### PR TITLE
[SPARK-15617][ML][DOC] Clarify that fMeasure in MulticlassMetrics is "micro" f1_score

### PR DESCRIPTION
**Pull request by zhengruifeng**
## What changes were proposed in this pull request?

1, del precision,recall in  `ml.MulticlassClassificationEvaluator`
2, update user guide for `mlllib.weightedFMeasure`
## How was this patch tested?

local build

### CODE REVIEW

File: mllib/src/main/scala/org/apache/spark/mllib/evaluation/MulticlassMetrics.scala [hunk 1]
```diff
@@ -153,7 +153,7 @@ class MulticlassMetrics @Since("1.1.0") (predictionAndLabels: RDD[(Double, Doubl
   lazy val recall: Double = accuracy
 
   /**
-   * Returns f-measure
+   * Returns micro averaged f1-measure
```
Comment (srowen) [inline mllib/src/main/scala/org/apache/spark/mllib/evaluation/MulticlassMetrics.scala]: There's more to it than just writing the name. It's accurate but not widely known, so this and the docs above should at least include a link and minimal explanation of what it means.

Comment (zhengruifeng) [inline mllib/src/main/scala/org/apache/spark/mllib/evaluation/MulticlassMetrics.scala]: Thanks, I will detail it

**Comment by zhengruifeng** [comment]
test this please

File: docs/mllib-evaluation-metrics.md [hunk 2]
```diff
@@ -182,17 +183,22 @@ $$\hat{\delta}(x) = \begin{cases}1 & \text{if $x = 0$}, \\ 0 & \text{otherwise}.
       </td>
     </tr>
     <tr>
-      <td>Overall Precision</td>
+      <td>Micro Accuracy</td>
```
Comment (srowen) [inline docs/mllib-evaluation-metrics.md]: "Micro" doesn't apply to accuracy, as I understand the definition.

File: docs/mllib-evaluation-metrics.md [hunk 3]
```diff
@@ -182,17 +183,22 @@ $$\hat{\delta}(x) = \begin{cases}1 & \text{if $x = 0$}, \\ 0 & \text{otherwise}.
...
+      <td>$ACC = \frac{TP}{TP + FP} = \frac{1}{N}\sum_{i=0}^{N-1} \hat{\delta}\left(\hat{\mathbf{y}}_i -
+        \mathbf{y}_i\right)$</td>
+    </tr>
+    <tr>
+      <td>Micro Precision</td>
```
Comment (srowen) [inline docs/mllib-evaluation-metrics.md]: Nit: I might call this "(Micro) Precision" to deemphasize the modifier. I personally had never heard it, and it may be a sklearn-specific description (?)  So maybe this makes it more readily understandable as "yes, just the definition you probably expect"

File: mllib/src/main/scala/org/apache/spark/ml/evaluation/MulticlassClassificationEvaluator.scala [hunk 4]
```diff
@@ -41,6 +41,10 @@ class MulticlassClassificationEvaluator @Since("1.5.0") (@Since("1.5.0") overrid
   /**
    * param for metric name in evaluation (supports `"f1"` (default), `"precision"`, `"recall"`,
    * `"weightedPrecision"`, `"weightedRecall"`, `"accuracy"`)
+   * Note: `"f1"` here means weighted averaged f1-measure, while the `fMeasure` in
```
Comment (srowen) [inline mllib/src/main/scala/org/apache/spark/ml/evaluation/MulticlassClassificationEvaluator.scala]: This applies to f1, precision and recall equally

File: docs/mllib-evaluation-metrics.md [hunk 5]
```diff
@@ -136,6 +136,7 @@ for all other classes. So, a true positive occurs whenever the prediction and th
 occurs when neither the prediction nor the label take on the value of a given class. By this convention, there can be
 multiple true negatives for a given data sample. The extension of false negatives and false positives from the former
 definitions of positive and negative labels is straightforward.
+Note that the [micro averaged](http://scikit-learn.org/stable/modules/generated/sklearn.metrics.f1_score.html) accuracy, precision, recall, and F1-measure are the same.
```
Comment (srowen) [inline docs/mllib-evaluation-metrics.md]: I think I might leave this out. Actually, overall precision/recall/F1 for multiclass are deprecated anyway.

File: docs/mllib-evaluation-metrics.md (hunk 3 above)
Comment (srowen) [inline docs/mllib-evaluation-metrics.md]: Hm, on second thought, since we've deprecated precision/recall/F1 for the not-by-label case maybe these should just be removed. Accuracy should stay.

File: mllib/src/main/scala/org/apache/spark/ml/evaluation/MulticlassClassificationEvaluator.scala (hunk 4 above)
Comment (srowen) [inline mllib/src/main/scala/org/apache/spark/ml/evaluation/MulticlassClassificationEvaluator.scala]: Also, fMeasure isn't a supported value here? or do I miss it? f1 is a measure. Here again I wonder if it's worth documenting it, since f1 / precision / recall are deprecated.

File: docs/mllib-evaluation-metrics.md (hunk 3 above)
Comment (zhengruifeng) [inline docs/mllib-evaluation-metrics.md]: The term `Micro Precision` is already used in the part of Multilabel classification, so I rename it here.
I agreed that we remove `precision/recall/F1` here.

File: docs/mllib-evaluation-metrics.md (hunk 2 above)
Comment (zhengruifeng) [inline docs/mllib-evaluation-metrics.md]: Yes, I will drop 'Micro' here

**Comment by zhengruifeng** [comment]
cc @srowen

File: mllib/src/main/scala/org/apache/spark/ml/evaluation/MulticlassClassificationEvaluator.scala [hunk 6]
```diff
@@ -39,16 +39,16 @@ class MulticlassClassificationEvaluator @Since("1.5.0") (@Since("1.5.0") overrid
   def this() = this(Identifiable.randomUID("mcEval"))
 
   /**
-   * param for metric name in evaluation (supports `"f1"` (default), `"precision"`, `"recall"`,
-   * `"weightedPrecision"`, `"weightedRecall"`, `"accuracy"`)
+   * param for metric name in evaluation (supports `"f1"` (default), `"weightedPrecision"`,
```
Comment (srowen) [inline mllib/src/main/scala/org/apache/spark/ml/evaluation/MulticlassClassificationEvaluator.scala]: ah, so f1 = weightedF1Measure? OK

**Comment by srowen** [comment]
That LGTM

File: mllib/src/main/scala/org/apache/spark/ml/evaluation/MulticlassClassificationEvaluator.scala (hunk 6 above)
Comment (zhengruifeng) [inline mllib/src/main/scala/org/apache/spark/ml/evaluation/MulticlassClassificationEvaluator.scala]: OK, I will rename `f1` to `weightedF1Measure`

**Comment by srowen** [comment]
Merged to master/2.0
